sklearn/neighbors/lof.py

```diff
@@ -0,0 +1,315 @@
+        pair of instances (rows) and the resulting value recorded. The callable
+        should take two arrays as input and return one value indicating the
+        distance between them. This works for Scipy's metrics, but is less
+        efficient than passing the metric name as a string.
+
+        Distance matrices are not supported.
+
+        Valid values for metric are:
+
+        - from scikit-learn: ['cityblock', 'cosine', 'euclidean', 'l1', 'l2',
+          'manhattan']
+
+        - from scipy.spatial.distance: ['braycurtis', 'canberra', 'chebyshev',
+          'correlation', 'dice', 'hamming', 'jaccard', 'kulsinski',
+          'mahalanobis', 'matching', 'minkowski', 'rogerstanimoto',
+          'russellrao', 'seuclidean', 'sokalmichener', 'sokalsneath',
+          'sqeuclidean', 'yule']
+
+        See the documentation for scipy.spatial.distance for details on these
+        metrics:
+        http://docs.scipy.org/doc/scipy/reference/spatial.distance.html
+
+    metric_params : dict, optional (default = None)
+        Additional keyword arguments for the metric function.
+
+    contamination : float in (0., 0.5), optional (default=0.1)
+        The amount of contamination of the data set, i.e. the proportion
+        of outliers in the data set. When fitting it is used to define the
+        threshold on the decision function.
+
+    n_jobs : int, optional (default = 1)
+        The number of parallel jobs to run for neighbors search.
+        If ``-1``, then the number of jobs is set to the number of CPU cores.
+        Affects only :meth:`kneighbors` and :meth:`kneighbors_graph` methods.
+
+
+    Attributes
+    ----------
+    outlier_factor_ : numpy array, shape (n_samples,)
+        The LOF of X. The lower, the more normal.
+        Inliers tend to have a LOF score close to 1, while outliers tend
+        to have a larger LOF score.
+
+        The local outlier factor (LOF) of a sample captures its
+        supposed 'degree of abnormality'.
+        It is the average of the ratio of the local reachability density of
+        a sample and those of its k-nearest neighbors.
+
+    References
+    ----------
+    .. [1] Breunig, M. M., Kriegel, H. P., Ng, R. T., & Sander, J. (2000, May).
+           LOF: identifying density-based local outliers. In ACM sigmod record.
+    """
+    def __init__(self, n_neighbors=5, algorithm='auto', leaf_size=30,
+                 metric='minkowski', p=2, metric_params=None,
+                 contamination=0.1, n_jobs=1):
+        self._init_params(n_neighbors=n_neighbors,
+                          algorithm=algorithm,
+                          leaf_size=leaf_size, metric=metric, p=p,
+                          metric_params=metric_params, n_jobs=n_jobs)
+
+        self.contamination = contamination
+
+    def fit_predict(self, X, y=None):
+        """Compute the local outlier factor (LOF) on X.
+
+        Return the labels (1 inlier, -1 outlier) of X according to LOF score
```

Comment:
Usually the first line is a summary and the text below is a longer explanation.
Currently they list two somewhat different things. I would say "Fits the model to the training set and returns the labels (...) on the training set according to the LOF score." Maybe add "The result of ``fit_predict`` is different from ``fit().predict()`` because...